Content.Shared/Whitelist/EntityWhitelist.cs

```diff
@@ -59,4 +60,52 @@ public sealed partial class EntityWhitelist
     /// </summary>
     [DataField]
     public bool RequireAll;
+
+    public override bool Equals(object? obj)
+    {
+        if (obj is EntityWhitelist whitelist)
+            return Equals(whitelist);
+        return false;
+    }
+
+    public bool Equals(EntityWhitelist? other)
+    {
+        if (ReferenceEquals(this, other)) return true;
+        if (other is null) return false;
+
+        if (RequireAll != other.RequireAll) return false;
+        if (!NullableSequenceEqual(Components, other.Components)) return false;
+        if (!NullableSequenceEqual(Sizes, other.Sizes)) return false;
+        if (!NullableSequenceEqual(Tags, other.Tags)) return false;
+
+        return true;
+    }
```

Comment:
This equals check surely shouldn't care about the order of entries in e.g. `Sizes`?